scripts/miniapps/Nekbone.md

```diff
@@ -0,0 +1,124 @@
+# Instruction for building and running Nekbone
+
+Implementation: Fortran, C, MPI (single threaded)
+
+Useful links:
+- CESAR Thermal Hydraulics web site: https://cesar.mcs.anl.gov/content/software/thermal_hydraulics
+
+## How to build
+
+1. Download Nekbone from
+https://cesar.mcs.anl.gov/content/software/thermal_hydraulics.  You
+must register to get a copy of the source code.  There is a top-level
+readme.pdf with build and run instructions.
+
+2. Load necessary PrgEnv module, set PATH and LD_LIBRARY_PATH and set
+recommended envrionment variables, e.g. for MPICH, etc.
+
+3. Modify make scipt `makenek` in the directory `test/example1`
+
+   Set source directory to where the Nekbone source was downloaded, e.g.
+   ```
+   # source path 
+   #SOURCE_ROOT="$HOME/nekbone-3.0/src" 
+   SOURCE_ROOT="/path/to/nekbone/nekbone-3.1/src"
+   ```
+
+   If using the Intel compiler and building for KNL, specify automatic
+   cpu dispatch to allow the executable to be run on KNL and non-KNL.
+
+   For the Fortran compilation:
+   ```
+   #F77="mpif77"
+   F77="mpif77 -axMIC-AVX512"
+   ```
+
+   For the C compilation:
+   ```
+   #CC="mpicc"
+   CC="mpicc -axMIC-AVX512"
+   ```
+   
+4. The build is problem-size specific, so you must modify the `SIZE`
+file. Increase the maximum number of ranks from 10 to 8192 to allow
+more flexibility in how to run the programm later.
+   ```
+   C     parameter (lp = 10)                     ! max number of processors
+         parameter (lp = 8192)                   ! max number of processors
+   ```
+
+5. Build
+   ```
+   % ./makenek ex1
+   ```
+   
+   This creates an executable called `nekbone` in the same directory.
+
+## Example runs
+
+For each example, the number of nodes used can be varied depending on
+whether Xeon or KNL nodes are used.  The problem is built into the
+executable and there are no parameters to provide or vary. Nekbone is
+a single-threaded application, so aprun launch parameters can easily
```

Comment:
Change "there are no parameters to provide or vary" to
"there are no parameters to allow you to vary the problem size".